geowebcache/wmts/src/test/java/org/geowebcache/service/wmts/WMTSRestTest.java

```diff
@@ -192,13 +197,15 @@ public void testGetTileJSONWithoutStyle() throws Exception {
         // Checking the response contains a tileUrl without the style
         assertTrue(
                 content.contains(
-                        "\"tiles\":[\"http://localhost/service/wmts/rest/mockLayerTileJSON/EPSG:900913/EPSG:900913:{z}/{y}/{x}?format=image/png\"]"));
+                        "\"tiles\":[\"http://localhost/service/wmts/rest/mockLayerTileJSON/EPSG:900913/EPSG:900913:{z}/{y}/{x}?format="
```

Comment:
Guess it cannot be done in GWC, but downstreams in GeoServer, it would be useful to compare the tile URLs and make sure they end up being different depending on what format parameter has been provided asking for the TileJSON, e.g., point to mvt if the tilejson was requested for mvt tiles, and to pngs if the tilejson was requested for png tiles (on a layer that supports both formats)

Reply:
I managed to have it done in a GWC Test. Updating it.